app/src/main/java/com/weather/weathercheckerapp/ViewModelFactory.kt

```diff
@@ -0,0 +1,25 @@
+package com.weather.weathercheckerapp
+
+import androidx.lifecycle.ViewModel
+import androidx.lifecycle.ViewModelProvider
+import javax.inject.Inject
+import javax.inject.Provider
+import javax.inject.Singleton
+
+@Suppress("UNCHECKED_CAST")
+@Singleton
+class ViewModelFactory @Inject constructor(private val viewModelsMap: Map<Class<out ViewModel>, @JvmSuppressWildcards Provider<ViewModel>>) : ViewModelProvider.Factory {
+
+    override fun <T : ViewModel> create(modelClass: Class<T>): T {
+        val creator = viewModelsMap[modelClass] ?:
+        viewModelsMap.asIterable().firstOrNull {
+            modelClass.isAssignableFrom(it.key)
+        }?.value ?: throw IllegalArgumentException("unknown model class $modelClass")
+        return try {
+            creator.get() as T
+        } catch (e: Exception) {
+            throw RuntimeException(e)
+        }
+    }
+
+}
```

Comment:
Missing new line, it is usually a good practice to end every file with an empty line, git can handle the files in that case better.